crates/engine/tree/src/tree/metrics.rs

```diff
@@ -88,6 +88,22 @@ impl BlockValidationMetrics {
     }
 }
 
+/// Metrics for prewarming threads
+#[derive(Metrics, Clone)]
+#[metrics(scope = "sync.prewarm")]
+pub(crate) struct PrewarmThreadMetrics {
+    /// Prewarm thread spawn duration
+    pub(crate) spawn_duration: Gauge,
+    /// The number of transactions in the block
+    pub(crate) transactions: Gauge,
```

Comment:
should we make these histograms too? In a low block time world Prometheus will not be able to fetch all values, so some of them will be lost

with histogram we'll have all

Reply:
Yeah, that would be useful

Reply:
added